### PR TITLE
clustermesh docs: add global and shared services reference

### PR DESCRIPTION
**Pull request by giorio94**
This is a followup of #23298, which adds a section to the cluster-mesh load-balancing and service discovery documentation, including a reference flow chart describing the behavior considering different combinations of the global-service and shared-service annotation values. The flow chart is a slightly adapted version of that proposed by @YutaroHayakawa (https://github.com/cilium/cilium/pull/23298#discussion_r1087433845), changing *Local* and *Remote* to *Cluster1* and *Cluster2* to reduce the possible confusion when describing the endpoints (i.e., to avoid referring to both to the clusters themselves and the location of the endpoints with the same terms).

The mermaid flow chart has been generated separately and imported in the documentation as an SVG file (the code is attached as a comment). Although suboptimal, the direct usage of sphinxcontrib-mermaid to generate it in the docs proved to be problematic, and would lead to the addition of multiple dependencies to the builder image.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

```release-note
clustermesh docs: add global and shared services reference
```

### CODE REVIEW

**Comment by giorio94** [comment]
TravisCI is failing due to https://github.com/cilium/cilium/issues/23314

File: Documentation/network/clustermesh/services.rst [hunk 1]
```diff
@@ -110,3 +110,32 @@ Deploying a Simple Example Service
...
+      Cluster1Global{Cluster1\nGlobal?}-->|yes|Cluster2Global{Cluster2\nGlobal?}
+      Cluster2Global-->|yes|Cluster1Shared{Cluster1\nShared?}
+
+      Cluster1Shared-->|yes|Cluster2Shared{Cluster2\nShared?}
+      Cluster2Shared-->|yes|Cluster1BothCluster2Both[Cluster1: Local + Remote\nCluster2: Local + Remote]
+      Cluster2Shared-->|no|Cluster1SelfClusterBoth[Cluster1: Local only\nCluster2: Local + Remote]
+
+      Cluster1Shared-->|no|Cluster2Shared2{Cluster2\nShared?}
+      Cluster2Shared2-->|yes|Cluster1BothCluster2Self[Cluster1: Local + Remote\nCluster2: Local only]
+      Cluster2Shared2-->|no|Cluster1SelfCluster2Self[Cluster1: Local only,\nCluster2: Local only]
```
Comment (YutaroHayakawa) [inline Documentation/network/clustermesh/services.rst]: Nit
```suggestion
      Cluster2Shared2-->|no|Cluster1SelfCluster2Self[Cluster1: Local only\nCluster2: Local only]
```

**Review by YutaroHayakawa — APPROVED** [review]
Great! Thanks!

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Review by qmonnet — APPROVED** [review]
Looks good, thank you!

I see the blocking label. What is this PR blocked on?

**Comment by giorio94** [comment]
> I see the blocking label. What is this PR blocked on?

I've removed the blocking label. It was there mainly to wait for https://github.com/cilium/cilium/pull/23298, which uniformed this behavior.

**Comment by borkmann** [comment]
(Marking for backport given https://github.com/cilium/cilium/pull/23298 is marked too.)

**Pull request merged**